test/object-store/util/sync/baas_admin_api.cpp

```diff
@@ -445,17 +445,20 @@ nlohmann::json AdminAPIEndpoint::patch_json(nlohmann::json body) const
     return nlohmann::json::parse(resp.body.empty() ? "{}" : resp.body);
 }
 
-AdminAPISession AdminAPISession::login(const std::string& base_url, const std::string& username,
-                                       const std::string& password)
+AdminAPISession AdminAPISession::login(const AppCreateConfig& config)
 {
+    std::string admin_url = config.admin_url;
     nlohmann::json login_req_body{
         {"provider", "userpass"},
```

Comment:
nit: should this be part of AppCreateConfig too?

Reply:
It already is or are you questioning whether it should be?
I am making a copy here from the AppCreateConfig structure so we don't lose the value when it is "moved" into the `AdminAPISession` object when that object is created.

Reply:
I was suggesting that it should be. Right now it is hardcoded.

Reply:
I don't think I follow... it is already part of the `AppCreateConfig` structure:
```c++
AppCreateConfig default_app_config() // the same for minimal_app_config()
{
    . . .
    std::string app_url = get_base_url();
    std::string admin_url = get_admin_url();
    . . .
    return AppCreateConfig{
        "test",
        std::move(app_url),
        std::move(admin_url), // BAAS Admin API URL may be different
        . . .
    };
}
```

Reply:
Maybe we are talking about different things. `admin_username` and `admin_password` are part of AppCreateConfig (they are used to create `login_req_body`), but the value for `provider` is hardcoded to `userpass`. My suggestion was to move the field to AppCreateConfig and give it a default value.

Reply:
Ah, sorry - I totally missed the line that you were referring to.
It is currently only configured for `userpass` in `install_baas.sh` and it would take additional work to support anything else.
Since I doubt it's going to change in the near term, I prefer to delay that change until other login methods are added.

Reply:
sounds good